processor/batchprocessor/metrics.go

```diff
@@ -76,3 +99,154 @@ func MetricViews() []*view.View {
 		distributionBatchSendSizeBytesView,
 	}
 }
+
+func OtelMetricsViews() ([]otelview.View, error) {
+	var views []otelview.View
+	var err error
+
+	v, err := otelview.New(
+		otelview.MatchInstrumentName(obsreport.BuildProcessorCustomMetricName(typeStr, "batch_send_size")),
+		otelview.WithSetAggregation(aggregation.ExplicitBucketHistogram{
+			Boundaries: []float64{10, 25, 50, 75, 100, 250, 500, 750, 1000, 2000, 3000, 4000, 5000, 6000, 7000, 8000, 9000, 10000, 20000, 30000, 50000, 100000},
+		}),
+	)
+	if err != nil {
+		return nil, err
+	}
+	views = append(views, v)
+
+	v, err = otelview.New(
+		otelview.MatchInstrumentName(obsreport.BuildProcessorCustomMetricName(typeStr, "batch_send_size_bytes")),
+		otelview.WithSetAggregation(aggregation.ExplicitBucketHistogram{
+			Boundaries: []float64{10, 25, 50, 75, 100, 250, 500, 750, 1000, 2000, 3000, 4000, 5000, 6000, 7000, 8000, 9000, 10000, 20000, 30000, 50000,
+				100_000, 200_000, 300_000, 400_000, 500_000, 600_000, 700_000, 800_000, 900_000,
+				1000_000, 2000_000, 3000_000, 4000_000, 5000_000, 6000_000, 7000_000, 8000_000, 9000_000},
+		}),
+	)
+	if err != nil {
+		return nil, err
+	}
+	views = append(views, v)
+
+	return views, nil
+}
+
+type batchProcessorTelemetry struct {
+	level    configtelemetry.Level
+	detailed bool
+	useOtel  bool
+
+	exportCtx context.Context
+
+	processorAttr        []attribute.KeyValue
+	batchSizeTriggerSend syncint64.Counter
+	timeoutTriggerSend   syncint64.Counter
+	batchSendSize        syncint64.Histogram
+	batchSendSizeBytes   syncint64.Histogram
+}
+
+func newBatchProcessorTelemetry(mp metric.MeterProvider, cfg *Config, level configtelemetry.Level, registry *featuregate.Registry) (*batchProcessorTelemetry, error) {
+	exportCtx, err := tag.New(context.Background(), tag.Insert(processorTagKey, cfg.ID().String()))
+	if err != nil {
+		return nil, err
+	}
+
+	bpt := &batchProcessorTelemetry{
+		useOtel:       registry.IsEnabled(obsreportconfig.UseOtelForInternalMetricsfeatureGateID),
+		processorAttr: []attribute.KeyValue{attribute.String(obsmetrics.ProcessorKey, cfg.ID().String())},
+		exportCtx:     exportCtx,
+		level:         level,
+		detailed:      level == configtelemetry.LevelDetailed,
+	}
+
+	err = bpt.createOtelMetrics(mp)
+	if err != nil {
+		return nil, err
+	}
+
+	return bpt, nil
+}
+
+func (bpt *batchProcessorTelemetry) createOtelMetrics(mp metric.MeterProvider) error {
+	if !bpt.useOtel {
+		return nil
+	}
+
+	var err error
+	meter := mp.Meter(scopeName)
+
+	bpt.batchSizeTriggerSend, err = meter.SyncInt64().Counter(
+		obsreport.BuildProcessorCustomMetricName(typeStr, "batch_size_trigger_send"),
+		instrument.WithDescription("Number of times the batch was sent due to a size trigger"),
+		instrument.WithUnit(unit.Dimensionless),
+	)
+	if err != nil {
+		return err
```

Comment:
what conditions would cause an error here and are they worth testing? only asking since it looks like many error conditions are not being tested

Reply:
From the otel-go docs:
```
// If the passed instrument would result in an incompatible Aggregator, an
// error is returned and that Aggregator is not inserted or returned.
```

I think for most cases this should not happen, and can only be a problem when using views that would generate incompatible aggregators.